### PR TITLE
doc/howto/install: add pre-deployment requirements

### PR DESCRIPTION
**Pull request by minaelee**
Pre-deployment requirements including general requirements, testing/development requirements, and production requirements added to [How to install MicroCloud](https://canonical-microcloud.readthedocs-hosted.com/en/latest/microcloud/how-to/install/) docs.

### CODE REVIEW

File: doc/how-to/install.md [hunk 1]
```diff
@@ -1,6 +1,76 @@
...
+  - 3 members
+  - For critical deployments, we recommend a minimum of 4 members
+- Memory:
+  - Minimum 32 GiB RAM per cluster member
+- Software: 
+  - For production deployments subscribed to Ubuntu Pro, each cluster member must use a LTS version of Ubuntu
+- Networking: 
+  - For each cluster member, we recommend dual-port network cards with a minimum 10 GiB capacity, or higher if low latency is essential
+- Storage:
+  - For each cluster member, we recommend at least 3 NVMe disks:
```
Comment (roosterfish) [inline doc/how-to/install.md:63]: You need a new entry in the `doc/.custom_wordlist.txt` to pass the spell checker for `NVMe`.

**Review by roosterfish — CHANGES_REQUESTED** [review]

File: doc/how-to/install.md (hunk 1 above)
Comment (minaelee) [reply doc/how-to/install.md:63]: done!

File: doc/how-to/install.md [hunk 2]
```diff
@@ -1,6 +1,76 @@
...
+A physical or virtual machine intended for use as a MicroCloud cluster member must meet the following prerequisites:
+
+- Software: 
+  - Ubuntu 22.04 or newer (LTS version recommended)
+  - snapd 2.59 or newer
+
+- Networking:
+  - Fixed IP addresses (DHCP not supported)
+  - Two network interfaces per cluster member, one for intra-cluster communication and one for external connectivity to the uplink network
+    - Partially or fully aggregated networking setups require more interfaces; see: {ref}`howto-ceph-networking`
```
Comment (roosterfish) [inline doc/how-to/install.md:21]: `fully disaggregated`?

File: doc/how-to/install.md [hunk 3]
```diff
@@ -1,6 +1,76 @@
...
+```
+
+```{group-tab} Production environments
+
+- Physical machines only (no VMs)
+- Minimum cluster size:
+  - 3 members
+  - For critical deployments, we recommend a minimum of 4 members
+- Memory:
+  - Minimum 32 GiB RAM per cluster member
```
Comment (roosterfish) [inline doc/how-to/install.md:57]: I guess this is mainly coming from Ceph? Should we mention this or even link to https://docs.ceph.com/en/latest/start/hardware-recommendations/#hardware-recommendations.

Comment (roosterfish) [reply doc/how-to/install.md:57]: Should the 8GiB statement from https://canonical-microcloud.readthedocs-hosted.com/en/latest/microcloud/reference/ be removed now as it's stated here?

Maybe on the `reference/` page we can link to the Ceph doc as on there I think the idea is to be more specific.

**Review by roosterfish — CHANGES_REQUESTED** [review]
Looking good, only a few remarks.

File: doc/how-to/install.md (hunk 3 above)
Comment (minaelee) [reply doc/how-to/install.md:57]: > Should the 8GiB statement from https://canonical-microcloud.readthedocs-hosted.com/en/latest/microcloud/reference/ be removed now as it's stated here?

Yes, I will update that page separately. It needs several updates.

> Maybe on the `reference/` page we can link to the Ceph doc as on there I think the idea is to be more specific.
The reference/ page already links to the Ceph docs.

File: doc/how-to/install.md (hunk 2 above)
Comment (minaelee) [reply doc/how-to/install.md]: just making sure you're paying attention :)
fixed!

File: doc/how-to/install.md [hunk 4]
```diff
@@ -1,6 +1,77 @@
...
+````{tabs}
+```{group-tab} General
+
+The requirements in this section apply to all MicroCloud deployments.
+
+A physical or virtual machine intended for use as a MicroCloud cluster member must meet the following prerequisites:
+
+- Software: 
+  - Ubuntu 22.04 or newer (LTS version recommended)
+    - If you intend to use ZFS storage, use a non-HWE (Hardware Enabled) variant of Ubuntu 22.04
```
Comment (roosterfish) [inline doc/how-to/install.md:16]: Looks HWE is missing in the wordlist.

**Review by roosterfish — APPROVED** [review]

**Pull request merged**